diff -Nur sane-backends-1.0.5.orig/Makefile.in sane-backends-1.0.5/Makefile.in
--- sane-backends-1.0.5.orig/Makefile.in	Sat May 26 12:42:25 2001
+++ sane-backends-1.0.5/Makefile.in	Mon Jul  2 13:56:35 2001
@@ -7,6 +7,7 @@
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
+DESTDIR =
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
@@ -48,10 +49,10 @@
 lsm: $(PACKAGE)-$(VERSION).lsm
 
 install: install-recursive
-	$(MKDIR) $(includedir)/sane
+	$(MKDIR) $(DESTDIR)$(includedir)/sane
 	@list="$(INSTALLED_INCLUDES)"; for file in $$list; do \
 	  echo installing $${file} in $(includedir)/sane...; \
-	  $(INSTALL_DATA) $${file} $(includedir)/sane || exit 1; \
+	  $(INSTALL_DATA) $${file} $(DESTDIR)$(includedir)/sane || exit 1; \
 	done
 
 uninstall: uninstall-recursive
diff -Nur sane-backends-1.0.5.orig/backend/Makefile.in sane-backends-1.0.5/backend/Makefile.in
--- sane-backends-1.0.5.orig/backend/Makefile.in	Sat Jun  9 21:54:30 2001
+++ sane-backends-1.0.5/backend/Makefile.in	Mon Jul  2 14:03:24 2001
@@ -1,4 +1,5 @@
 SHELL = /bin/sh
+DESTDIR =
 
 VPATH = @srcdir@
 srcdir = @srcdir@
@@ -91,16 +92,16 @@
 all:	$(LIBS) libsane.la
 
 install:
-	$(MKDIR) $(libdir) $(libsanedir) $(configdir)
+	$(MKDIR) $(DESTDIR)$(libdir) $(DESTDIR)$(libsanedir) $(DESTDIR)$(configdir)
 	@list="$(LIBS)"; for be in $$list; do \
 	  echo installing $${be} in $(libsanedir)...; \
-	  $(LIBTOOL) $(MINST) $(INSTALL_PROGRAM) $${be} $(libsanedir)/$${be} \
+	  $(LIBTOOL) $(MINST) $(INSTALL_PROGRAM) $${be} $(DESTDIR)$(libsanedir)/$${be} \
 	    || exit 1; \
 	done
-	@$(LIBTOOL) $(MINST) --finish $(libsanedir)
+	@$(LIBTOOL) $(MINST) --finish $(DESTDIR)$(libsanedir)
 	@# Assume the dll name without any versions is last
 	@dllend=`../tools/libtool-get-dll-ext libsane-dll.la`; \
-	list="$(ALL_BACKENDS)"; cd $(libsanedir) && for be in $$list; do \
+	list="$(ALL_BACKENDS)"; cd $(DESTDIR)$(libsanedir) && for be in $$list; do \
 	  file=libsane-$${be}.$$dllend.$(V_MAJOR); \
 	  lib=`grep dlname= libsane-$${be}.la | cut -f2 -d"'"`; \
 	  if test ! -f $${file} -a -n "$${lib}"; then \
@@ -109,22 +110,19 @@
 	done
 	@# Assume the dll name without any versions is last
 	@dllend=`../tools/libtool-get-dll-ext libsane-dll.la`; \
-	rm -f $(libdir)/libsane.a $(libdir)/libsane.$$dllend \
-		$(libdir)/libsane.$$dllend.$(V_MAJOR)*; \
-	$(LN_S) sane/libsane-dll.a $(libdir)/libsane.a; \
-	$(LN_S) sane/libsane-dll.$$dllend $(libdir)/libsane.$$dllend ; \
-	cd $(libsanedir) && for n in libsane-dll.$$dllend.$(V_MAJOR)*; do \
+	rm -f $(DESTDIR)$(libdir)/libsane.a $(DESTDIR)$(libdir)/libsane.$$dllend \
+		$(DESTDIR)$(libdir)/libsane.$$dllend.$(V_MAJOR)*; \
+	$(LN_S) sane/libsane-dll.a $(DESTDIR)$(libdir)/libsane.a; \
+	$(LN_S) sane/libsane-dll.$$dllend $(DESTDIR)$(libdir)/libsane.$$dllend ; \
+	cd $(DESTDIR)$(libsanedir) && for n in libsane-dll.$$dllend.$(V_MAJOR)*; do \
 	  nn=`echo $$n | sed 's,^libsane-dll,libsane,'`; \
 	  (cd ..; $(LN_S) sane/$$n $$nn); \
 	done || exit 1
-	$(LIBTOOL) $(MINST) $(INSTALL_PROGRAM) libsane.la $(libdir)/libsane.la
+	$(LIBTOOL) $(MINST) $(INSTALL_PROGRAM) libsane.la $(DESTDIR)$(libdir)/libsane.la
 	@list="$(CONFIGS)"; for cfg in $$list; do \
-	  if test ! -r $(srcdir)/$${cfg}; then continue; fi; \
-	  if test -f $(configdir)/$${cfg}; then \
-	    echo NOT overwriting $${cfg} in $(configdir)...; \
-	  else \
+	  if test -f $(srcdir)/$${cfg}; then \
 	    echo installing $${cfg} in $(configdir)...; \
-	    $(INSTALL_DATA) $(srcdir)/$${cfg} $(configdir)/$${cfg} || exit 1; \
+	    $(INSTALL_DATA) $(srcdir)/$${cfg} $(DESTDIR)$(configdir)/$${cfg} || exit 1; \
 	  fi; \
 	done
 
diff -Nur sane-backends-1.0.5.orig/doc/Makefile.in sane-backends-1.0.5/doc/Makefile.in
--- sane-backends-1.0.5.orig/doc/Makefile.in	Sun Jun 10 20:20:55 2001
+++ sane-backends-1.0.5/doc/Makefile.in	Mon Jul  2 14:06:13 2001
@@ -1,4 +1,5 @@
 SHELL = /bin/sh
+DESTDIR =
 
 VPATH = @srcdir@
 srcdir = @srcdir@
@@ -67,37 +68,37 @@
 	@echo Generating manpage $@...
 
 install: all
-	$(MKDIR) $(mandir)/man1 $(mandir)/man5 $(mandir)/man7
+	$(MKDIR) $(DESTDIR)$(mandir)/man1 $(DESTDIR)$(mandir)/man5 $(DESTDIR)$(mandir)/man7
 	@for page in $(SECT1); do \
 	  echo installing $${page} in $(mandir)/man1/$${page}...; \
-	  $(INSTALL_DATA) $${page} $(mandir)/man1/$${page} || exit 1; \
+	  $(INSTALL_DATA) $${page} $(DESTDIR)$(mandir)/man1/$${page} || exit 1; \
 	done
 	@for page in $(SECT5); do \
 	  echo installing $${page} in $(mandir)/man5/$${page}...; \
-	  $(INSTALL_DATA) $${page} $(mandir)/man5/$${page} || exit 1; \
+	  $(INSTALL_DATA) $${page} $(DESTDIR)$(mandir)/man5/$${page} || exit 1; \
 	done
 	@for page in $(SECT7); do \
 	  echo installing $${page} in $(mandir)/man7/$${page}...; \
-	  $(INSTALL_DATA) $${page} $(mandir)/man7/$${page} || exit 1; \
+	  $(INSTALL_DATA) $${page} $(DESTDIR)$(mandir)/man7/$${page} || exit 1; \
 	done
-	$(MKDIR) $(docdir)
+	$(MKDIR) $(DESTDIR)$(docdir)
 	@for readme in $(READMES); do \
 	  echo installing $${readme} in $(docdir)/$${readme}...; \
-	  $(INSTALL_DATA) $(top_srcdir)/$${readme} $(docdir)/$${readme} \
+	  $(INSTALL_DATA) $(top_srcdir)/$${readme} $(DESTDIR)$(docdir)/$${readme} \
 	    || exit 1; \
 	done
 	@for backdir in $(BACKDIRS); do \
 	  echo installing $${backdir} in $(docdir)/$${backdir}...; \
-	  $(MKDIR) $(docdir)/$${backdir} ; \
+	  $(MKDIR) $(DESTDIR)$(docdir)/$${backdir} ; \
 	  for doc in $${backdir}/* ; do \
 	    if test -f $${doc} ; then \
-	      $(INSTALL_DATA) $${doc} $(docdir)/$${backdir} || exit 1; \
+	      $(INSTALL_DATA) $${doc} $(DESTDIR)$(docdir)/$${backdir} || exit 1; \
 	    fi \
 	  done \
 	done
 	@-for doc in $(DOCS); do \
 	    echo installing $${doc} in $(docdir)/$${doc}...; \
-	    $(INSTALL_DATA) $${doc} $(docdir)/$${doc}; \
+	    $(INSTALL_DATA) $${doc} $(DESTDIR)$(docdir)/$${doc}; \
 	  done
 
 uninstall:
@@ -148,22 +149,22 @@
 
 install-mostang: html sane.ps sane.ps.gz
 	@echo Installing html manpages in $(WWW_PATH)/man...
-	$(MKDIR) $(WWW_PATH)/man
+	$(MKDIR) $(DESTDIR)$(WWW_PATH)/man
 	for manpage in $(MANPAGES) ; do \
-	  $(INSTALL_DATA) $${manpage}.html $(WWW_PATH)/man/$${manpage}.html; \
+	  $(INSTALL_DATA) $${manpage}.html $(DESTDIR)$(WWW_PATH)/man/$${manpage}.html; \
 	done
 	@echo Installing sane-backends.html in $(WWW_PATH)...
-	$(MKDIR) $(WWW_PATH)
-	$(INSTALL_DATA) sane-backends.html $(WWW_PATH)/sane-backends.html
+	$(MKDIR) $(DESTDIR)$(WWW_PATH)
+	$(INSTALL_DATA) sane-backends.html $(DESTDIR)$(WWW_PATH)/sane-backends.html
 	@echo Installing SANE standard \(html\) in $(WWW_PATH)/html/...
-	$(MKDIR) $(WWW_PATH)/html/
+	$(MKDIR) $(DESTDIR)$(WWW_PATH)/html/
 	for html in sane/* ; do \
-	  $(INSTALL_DATA) $${html} $(WWW_PATH)/html/ || exit 1; \
+	  $(INSTALL_DATA) $${html} $(DESTDIR)$(WWW_PATH)/html/ || exit 1; \
 	done
 	@echo Installing SANE standard \(postscript\) in $(WWW_PATH)/...
-	$(INSTALL_DATA) sane.ps $(WWW_PATH)/sane.ps
+	$(INSTALL_DATA) sane.ps $(DESTDIR)$(WWW_PATH)/sane.ps
 	@echo Installing SANE standard \(ps.gz\) in $(WWW_PATH)/...
-	$(INSTALL_DATA) sane.ps.gz $(WWW_PATH)/sane.ps.gz
+	$(INSTALL_DATA) sane.ps.gz $(DESTDIR)$(WWW_PATH)/sane.ps.gz
 
 clean:
 	rm -f *.toc *.aux *.log *.cp *.fn *.tp *.vr *.pg *.ky *.blg *.idx *.cb
diff -Nur sane-backends-1.0.5.orig/frontend/Makefile.in sane-backends-1.0.5/frontend/Makefile.in
--- sane-backends-1.0.5.orig/frontend/Makefile.in	Sat May 26 10:32:35 2001
+++ sane-backends-1.0.5/frontend/Makefile.in	Mon Jul  2 14:07:46 2001
@@ -1,4 +1,5 @@
 SHELL = /bin/sh
+DESTDIR =
 
 VPATH = @srcdir@
 srcdir = @srcdir@
@@ -61,14 +62,14 @@
 all: $(PROGRAMS)
 
 install: $(PROGRAMS)
-	$(MKDIR) $(bindir) $(sbindir) $(datadir)
+	$(MKDIR) $(DESTDIR)$(bindir) $(DESTDIR)$(sbindir) $(DESTDIR)$(datadir)
 	@for program in $(BINPROGS); do \
 	  $(LIBTOOL) $(MINST) $(INSTALL_PROGRAM) $${program} \
-	    $(bindir)/$${program}; \
+	    $(DESTDIR)$(bindir)/$${program}; \
 	done
 	@for program in $(SBINPROGS); do \
 	  $(LIBTOOL) $(MINST) $(INSTALL_PROGRAM) $${program} \
-	    $(sbindir)/$${program}; \
+	    $(DESTDIR)$(sbindir)/$${program}; \
 	done
 
 uninstall:
diff -Nur sane-backends-1.0.5.orig/japi/Makefile.in sane-backends-1.0.5/japi/Makefile.in
--- sane-backends-1.0.5.orig/japi/Makefile.in	Sun Sep 17 05:02:24 2000
+++ sane-backends-1.0.5/japi/Makefile.in	Mon Jul  2 14:08:38 2001
@@ -3,6 +3,7 @@
 #
 
 SHELL = /bin/sh
+DESTDIR =
 
 VPATH = @srcdir@
 srcdir = @srcdir@
@@ -92,10 +93,10 @@
 #	Install library.  Don't know where to install .class files yet.
 
 install:
-	$(MKDIR) $(libdir) $(configdir)
+	$(MKDIR) $(DESTDIR)$(libdir) $(DESTDIR)$(configdir)
 	$(LIBTOOL) $(MINST) $(INSTALL_PROGRAM) libsanej.la \
-		$(libdir)/libsanej.la || exit 1;
-	@$(LIBTOOL) $(MINST) --finish $(libdir)
+		$(DESTDIR)$(libdir)/libsanej.la || exit 1;
+	@$(LIBTOOL) $(MINST) --finish $(DESTDIR)$(libdir)
 
 
 Sane.c: Sane.h
diff -Nur sane-backends-1.0.5.orig/tools/Makefile.in sane-backends-1.0.5/tools/Makefile.in
--- sane-backends-1.0.5.orig/tools/Makefile.in	Tue Jun  5 18:15:10 2001
+++ sane-backends-1.0.5/tools/Makefile.in	Mon Jul  2 14:10:11 2001
@@ -1,4 +1,5 @@
 SHELL = /bin/sh
+DESTDIR =
 
 VPATH = @srcdir@
 srcdir = @srcdir@
@@ -66,8 +67,8 @@
 all:	$(DESTINATIONS)
 
 install: sane-config sane-find-scanner
-	$(INSTALL_SCRIPT) sane-config $(bindir)/sane-config
-	$(INSTALL_PROGRAM) sane-find-scanner $(bindir)/sane-find-scanner
+	$(INSTALL_SCRIPT) sane-config $(DESTDIR)$(bindir)/sane-config
+	$(INSTALL_PROGRAM) sane-find-scanner $(DESTDIR)$(bindir)/sane-find-scanner
 
 uninstall: 
 	rm -f $(bindir)/sane-config $(bindir)/sane-find-scanner
