--- sane-backends-1.2.1/backend/Makefile.am.orig	2023-11-17 21:02:29.569878012 +0100
+++ sane-backends-1.2.1/backend/Makefile.am	2023-11-17 21:06:27.006433425 +0100
@@ -335,7 +335,7 @@
     sane_strstatus.lo \
     ../sanei/sanei_usb.lo \
     ../sanei/sanei_thread.lo \
-    $(MATH_LIB) $(USB_LIBS) $(SANEI_THREAD_LIBS) $(RESMEG_LIBS)
+    $(MATH_LIB) $(USB_LIBS) $(SANEI_THREAD_LIBS) $(RESMGR_LIBS)
 EXTRA_DIST += artec_eplus48u.conf.in
 
 libas6e_la_SOURCES = as6e.c as6e.h
@@ -1586,7 +1586,7 @@
     ../sanei/sanei_usb.lo \
     ../sanei/sanei_config.lo \
     sane_strstatus.lo \
-    $(USB_LIBS)
+    $(USB_LIBS) $(RESMGR_LIBS)
 EXTRA_DIST += ricoh2_buffer.c
 
 librts8891_la_SOURCES = rts8891.c rts8891.h rts88xx_lib.c rts88xx_lib.h
@@ -1966,11 +1966,11 @@
 libdll_preload_la_SOURCES =  dll.c
 libdll_preload_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=dll -DENABLE_PRELOAD
 libdll_preload_la_LIBADD = ../sanei/sanei_usb.lo \
-     $(USB_LIBS) $(XML_LIBS)
+     $(USB_LIBS) $(RESMGR_LIBS) $(XML_LIBS)
 libdll_la_SOURCES =  dll.c
 libdll_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=dll
 libdll_la_LIBADD = ../sanei/sanei_usb.lo \
-     $(USB_LIBS) $(XML_LIBS)
+     $(USB_LIBS) $(RESMGR_LIBS) $(XML_LIBS)
 BUILT_SOURCES += dll-preload.h
 CLEANFILES += dll-preload.h
 
--- sane-backends-1.0.29/tools/Makefile.am.orig	2020-04-19 08:47:07.661491206 +0200
+++ sane-backends-1.0.29/tools/Makefile.am	2020-04-19 08:49:11.890818198 +0200
@@ -42,7 +42,7 @@
 sane_find_scanner_SOURCES += check-usb-chip.c
 endif
 sane_find_scanner_LDADD = ../sanei/libsanei.la ../lib/liblib.la \
-                          $(USB_LIBS) $(IEEE1284_LIBS) $(SCSI_LIBS) $(XML_LIBS) \
+                          $(USB_LIBS) $(IEEE1284_LIBS) $(SCSI_LIBS) $(RESMGR_LIBS) $(XML_LIBS) \
 			  ../backend/sane_strstatus.lo
 
 gamma4scanimage_SOURCES = gamma4scanimage.c
