--- sane-backends-1.0.15/po/Makefile.in.orig	2004-10-14 18:44:17.000000000 +0200
+++ sane-backends-1.0.15/po/Makefile.in	2004-11-07 23:09:17.120050056 +0100
@@ -2,7 +2,7 @@
 #
 # See po/README for documentation.
 
-ALL_LINGUAS = bg cs da de es fi fr it nl no pl pt ru sv 
+ALL_LINGUAS = bg cs da de es fi fr it nl nb pl pt ru sv 
 
 SRC_FILES = ../include/sane/saneopts.h ../backend/artec_eplus48u.c \
   ../backend/avision.h \
