--- sane-backends-1.0.21/backend/Makefile.am.orig	2010-11-06 21:29:14.551101613 +0100
+++ sane-backends-1.0.21/backend/Makefile.am	2010-11-07 10:26:58.167102452 +0100
@@ -271,7 +271,7 @@ libartec_eplus48u_la_CPPFLAGS = $(AM_CPP
 nodist_libsane_artec_eplus48u_la_SOURCES = artec_eplus48u-s.c
 libsane_artec_eplus48u_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=artec_eplus48u
 libsane_artec_eplus48u_la_LDFLAGS = $(DIST_SANELIBS_LDFLAGS)
-libsane_artec_eplus48u_la_LIBADD = $(COMMON_LIBS) libartec_eplus48u.la ../sanei/sanei_init_debug.lo ../sanei/sanei_constrain_value.lo ../sanei/sanei_config.lo  sane_strstatus.lo ../sanei/sanei_usb.lo ../sanei/sanei_thread.lo $(MATH_LIB) $(USB_LIBS) $(PTHREAD_LIBS) $(RESMEG_LIBS)
+libsane_artec_eplus48u_la_LIBADD = $(COMMON_LIBS) libartec_eplus48u.la ../sanei/sanei_init_debug.lo ../sanei/sanei_constrain_value.lo ../sanei/sanei_config.lo  sane_strstatus.lo ../sanei/sanei_usb.lo ../sanei/sanei_thread.lo $(MATH_LIB) $(USB_LIBS) $(PTHREAD_LIBS) $(RESMGR_LIBS)
 EXTRA_DIST += artec_eplus48u.conf.in
 
 libas6e_la_SOURCES = as6e.c as6e.h

--- a/tools/Makefile.am~	2016-10-09 02:54:58.000000000 +0200
+++ b/tools/Makefile.am	2018-05-09 15:43:07.450971874 +0200
@@ -39,7 +39,7 @@ if have_usblib
 sane_find_scanner_SOURCES += check-usb-chip.c
 endif
 sane_find_scanner_LDADD = ../sanei/libsanei.la ../lib/liblib.la \
-			  $(USB_LIBS) $(IEEE1284_LIBS) $(SCSI_LIBS) \
+			  $(USB_LIBS) $(IEEE1284_LIBS) $(SCSI_LIBS) $(RESMGR_LIBS) \
 			  ../backend/sane_strstatus.lo
 
 gamma4scanimage_SOURCES = gamma4scanimage.c
