--- sane-backends-1.0.18/sane_hp_rts88xx/sane_hp_rts88xx/patch-sane.sh.orig	2004-11-21 14:28:11.000000000 +0100
+++ sane-backends-1.0.18/sane_hp_rts88xx/sane_hp_rts88xx/patch-sane.sh	2006-08-20 21:12:37.084515000 +0200
@@ -82,9 +82,14 @@
 sed "s/PRELOADABLE_BACKENDS =/& hp_rts88xx /g" $sane/backend/Makefile.in | \
     sed "s/DISTFILES = /& hp_rts88xx.c hp_rts88xx.h /g"  | \
         sed 's%libsane-abaton.la: ../sanei/sanei_scsi.lo%& \
-libsane-hp_rts88xx.la: ../sanei/sanei_config2.lo \
 libsane-hp_rts88xx.la: ../sanei/sanei_constrain_value.lo \
-libsane-hp_rts88xx.la: ../sanei/sanei_usb.lo%' >  $sane/backend/tmp~
+libsane-hp_rts88xx.la: ../sanei/sanei_usb.lo%' | sed '
+s%^libsane-artec_eplus48u\.la: artec_eplus48u\.lo%libsane-hp_rts88xx.la: hp_rts88xx.lo hp_rts88xx-s.lo $(EXTRA) $(LIBOBJS)\
+	@$(LIBTOOL) $(MLINK) $(CC) -export-dynamic -o $@ $($*_LIBS) @USB_LIBS@ @LIBRESMGR@ \\\
+	$(LDFLAGS) $(BACKENDLIBS) $^ -rpath $(libsanedir) \\\
+	-version-info $(V_MAJOR):$(V_REV):$(V_MINOR) $(DYNAMIC_FLAG)\
+\
+&%' >  $sane/backend/tmp~
 evaluate_retval
 
 echo -n "Relocating files..."
@@ -99,10 +98,10 @@
 evaluate_retval
 
 # Adds a line with 'hp_rts88xx' to dll.conf
-echo -n "Modifying backend/dll.conf..."
-if [ -z $(grep hp_rts88xx $sane/backend/dll.conf) ]
+echo -n "Modifying backend/dll.conf.in..."
+if [ -z $(grep hp_rts88xx $sane/backend/dll.conf.in) ]
     then
-    echo "hp_rts88xx" >> $sane/backend/dll.conf
+    echo "hp_rts88xx" >> $sane/backend/dll.conf.in
 fi
 evaluate_retval
 
--- sane-backends-1.0.18/sane_hp_rts88xx/sane_hp_rts88xx/hp_rts88xx.desc.orig	2004-11-21 12:32:09.000000000 +0100
+++ sane-backends-1.0.18/sane_hp_rts88xx/sane_hp_rts88xx/hp_rts88xx.desc	2006-08-20 21:17:49.436035750 +0200
@@ -13,25 +13,12 @@
 
 :model "HP4400C"                                      ; name models for above-specified mfg.
 :interface "USB"
+:usbid "0x03f0" "0x0705"
 :status :basic
 :comment "grayscale 300/600 DPI"
 
 :model "HP4470C"
 :interface "USB"
+:usbid "0x03f0" "0x0805"
 :status :basic
 :comment "grayscale 300/600 DPI, XPA/bed mode"
-
-:model "HP3500C"
-:interface "USB"
-:status :basic
-:comment "not tested"
-
-:model "HP3530C"
-:interface "USB"
-:status :basic
-:comment "not tested"
-
-:model "HP3570C"
-:interface "USB"
-:status :basic
-:comment "not tested"
