diff -Nur sane-backends-1.0.8.orig/acinclude.m4 sane-backends-1.0.8/acinclude.m4
--- sane-backends-1.0.8.orig/acinclude.m4	Sun Jun 16 18:23:57 2002
+++ sane-backends-1.0.8/acinclude.m4	Sun Jun 16 18:50:07 2002
@@ -101,16 +101,18 @@
 # Checks for ieee1284 library, needed for canon_pp backend.
 AC_DEFUN(SANE_CHECK_IEEE1284,
 [
+  IEEE1284_LIBS=""
   AC_CHECK_HEADER(ieee1284.h, [
     AC_CACHE_CHECK([for libieee1284 >= 0.1.5], sane_cv_use_libieee1284, [
       AC_TRY_COMPILE([#include <ieee1284.h>], [
 	struct parport p; char *buf; 
 	ieee1284_nibble_read(&p, 0, buf, 1);
  	], 
-        [sane_cv_use_libieee1284="yes"; LIBS="${LIBS} -lieee1284"
+        [sane_cv_use_libieee1284="yes"; IEEE1284_LIBS="-lieee1284"
       ],[sane_cv_use_libieee1284="no"])
     ],)
   ],)
+  AC_SUBST(IEEE1284_LIBS)
 ])
 
 
@@ -119,6 +121,7 @@
 # GPHOTO2 backends.
 AC_DEFUN(SANE_CHECK_JPEG,
 [
+  JPEG_LIBS=""
   AC_CHECK_LIB(jpeg,jpeg_start_decompress, 
   [
     AC_CHECK_HEADER(jconfig.h, 
@@ -130,10 +133,11 @@
         #if JPEG_LIB_VERSION >= 61
           sane_correct_jpeg_lib_version_found
         #endif
-      ],[sane_cv_use_libjpeg="yes"; LIBS="${LIBS} -ljpeg"; 
+      ],[sane_cv_use_libjpeg="yes"; JPEG_LIBS="-ljpeg";
       AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)])
     ],)
   ],)
+  AC_SUBST(JPEG_LIBS)
 ])
 
 dnl
@@ -237,7 +241,7 @@
 			CPPFLAGS="${CPPFLAGS} `gphoto2-config --cflags`"
 			GPHOTO2_LIBS="`gphoto2-config --libs`"
 			SANE_EXTRACT_LDFLAGS(LDFLAGS, GPHOTO2_LIBS)
-			LIBS="${LIBS} ${GPHOTO2_LIBS}"
+			AC_SUBST(GPHOTO2_LIBS)
 			;;
 		* ) 
 			echo 
diff -Nur sane-backends-1.0.8.orig/backend/Makefile.in sane-backends-1.0.8/backend/Makefile.in
--- sane-backends-1.0.8.orig/backend/Makefile.in	Mon May 27 21:19:55 2002
+++ sane-backends-1.0.8/backend/Makefile.in	Sun Jun 16 18:50:19 2002
@@ -127,6 +127,36 @@
 
 .PHONY: all clean depend dist distclean install uninstall
 
+libsane-canon_pp.la: canon_pp.lo canon_pp-s.lo $(EXTRA) $(LIBOBJS)
+	@$(LIBTOOL) $(MLINK) $(CC) -export-dynamic -o $@ $($*_LIBS) @IEEE1284_LIBS@ \
+	$(LDFLAGS) $(BACKENDLIBS) $^ -rpath $(libsanedir) \
+	-version-info $(V_MAJOR):$(V_REV):$(V_MINOR)
+
+libsane-dc210.la: dc210.lo dc210-s.lo $(EXTRA) $(LIBOBJS)
+	@$(LIBTOOL) $(MLINK) $(CC) -export-dynamic -o $@ $($*_LIBS) @JPEG_LIBS@ \
+	$(LDFLAGS) $(BACKENDLIBS) $^ -rpath $(libsanedir) \
+	-version-info $(V_MAJOR):$(V_REV):$(V_MINOR)
+
+libsane-dc240.la: dc240.lo dc240-s.lo $(EXTRA) $(LIBOBJS)
+	@$(LIBTOOL) $(MLINK) $(CC) -export-dynamic -o $@ $($*_LIBS) @JPEG_LIBS@ \
+	$(LDFLAGS) $(BACKENDLIBS) $^ -rpath $(libsanedir) \
+	-version-info $(V_MAJOR):$(V_REV):$(V_MINOR)
+
+libsane-dll.la: dll.lo dll-s.lo $(EXTRA) $(LIBOBJS)
+	@$(LIBTOOL) $(MLINK) $(CC) -export-dynamic -o $@ $($*_LIBS) @JPEG_LIBS@ \
+	$(LDFLAGS) $(BACKENDLIBS) $^ -rpath $(libsanedir) \
+	-version-info $(V_MAJOR):$(V_REV):$(V_MINOR)
+
+libsane-gphoto2.la: gphoto2.lo gphoto2-s.lo $(EXTRA) $(LIBOBJS)
+	@$(LIBTOOL) $(MLINK) $(CC) -export-dynamic -o $@ $($*_LIBS) @GPHOTO2_LIBS@ @JPEG_LIBS@ \
+	$(LDFLAGS) $(BACKENDLIBS) $^ -rpath $(libsanedir) \
+	-version-info $(V_MAJOR):$(V_REV):$(V_MINOR)
+
+libsane-sm3600.la: sm3600.lo sm3600-s.lo $(EXTRA) $(LIBOBJS)
+	@$(LIBTOOL) $(MLINK) $(CC) -export-dynamic -o $@ $($*_LIBS) -lusb \
+	$(LDFLAGS) $(BACKENDLIBS) $^ -rpath $(libsanedir) \
+	-version-info $(V_MAJOR):$(V_REV):$(V_MINOR)
+
 libsane-%.la: %.lo %-s.lo $(EXTRA) $(LIBOBJS)
 	@$(LIBTOOL) $(MLINK) $(CC) -export-dynamic -o $@ $($*_LIBS) \
 	$(LDFLAGS) $(BACKENDLIBS) $^ -rpath $(libsanedir) \
@@ -214,7 +244,7 @@
 
 libsane.la: dll.lo dll-s.lo $(EXTRA) $(addsuffix .lo,$(DLL_PRELOAD)) $(LIBOBJS)
 	@$(LIBTOOL) $(MLINK) $(CC) -o $@ $(LDFLAGS) $(BACKENDLIBS)  $^ \
-		$(addsuffix .lo,$(DLL_PRELOAD_EXTRAS)) \
+		$(addsuffix .lo,$(DLL_PRELOAD_EXTRAS)) @JPEG_LIBS@ \
 		-rpath $(libdir) -version-info $(V_MAJOR):$(V_REV):$(V_MINOR)
 
 # additional dependencies
diff -Nur sane-backends-1.0.8.orig/configure.in sane-backends-1.0.8/configure.in
--- sane-backends-1.0.8.orig/configure.in	Sun Jun 16 18:23:57 2002
+++ sane-backends-1.0.8/configure.in	Sun Jun 16 18:26:30 2002
@@ -206,7 +206,9 @@
 AC_CHECK_LIB(m,sqrt)
 AC_CHECK_LIB(scsi, scsireq_enter)	# FreeBSD needs this
 AC_CHECK_LIB(cam, cam_open_device)      # FreeBSD 3+ needs this
+save_LIBS="$LIBS"
 AC_CHECK_LIB(usb, usb_open)
+LIBS="$save_LIBS"
 
 SANE_CHECK_JPEG
 
