--- sane-1.0.2/sanei/sanei_ab306.c.orig	Thu May 18 18:22:45 2000
+++ sane-1.0.2/sanei/sanei_ab306.c	Thu May 18 18:24:27 2000
@@ -49,6 +49,10 @@
 
 #include <sys/types.h>
 
+#ifdef __sparc__
+#define IO_SUPPORT_MISSING
+#else
+
 #ifdef HAVE_SYS_IO_H
 # include <sys/io.h>	/* use where available (glibc 2.x, for example) */
 #elif HAVE_ASM_IO_H
@@ -72,6 +76,8 @@
 
 #else
 # define IO_SUPPORT_MISSING
+#endif
+
 #endif
 
 #include <sane/sane.h>
--- sane-1.0.2/sanei/sanei_pio.c.orig	Thu May 18 18:24:42 2000
+++ sane-1.0.2/sanei/sanei_pio.c	Thu May 18 18:25:31 2000
@@ -58,6 +58,10 @@
 # include <unistd.h>
 #endif
 
+#ifdef __sparc__
+#define IO_SUPPORT_MISSING
+#else
+
 #ifdef HAVE_SYS_IO_H
 # include <sys/io.h>	/* use where available (glibc 2.x, for example) */
 #elif HAVE_ASM_IO_H
@@ -83,6 +87,7 @@
 
 #else
 # define IO_SUPPORT_MISSING
+#endif
 #endif
 
 #include <sane/sane.h>
