--- sane-backends-1.0.23/configure.in.orig	2012-08-19 22:31:34.000000000 +0200
+++ sane-backends-1.0.23/configure.in	2013-05-28 18:03:13.483572312 +0200
@@ -331,12 +331,13 @@
 if test -e /usr/include/cups/cups.h ; then
     AC_DEFINE(HAVE_CUPS, 1, [Is /usr/include/cups/cups.h available?])
 	with_cups="yes"
-	LIBS="-lcups  $LIBS"
+	CUPS_LIBS="-lcups"
 else
 	$as_echo "cups.h not found, you may want to install a cups development package"
 	$as_echo "in order to autodetect network scanners in kodakaio."
 	with_cups="no"
 fi
+AC_SUBST([CUPS_LIBS])
 
 dnl ***********
 dnl USB Support
