--- sane-backends-1.0.6/backend/microtek2.c.microtek2	Tue Nov 20 09:14:24 2001
+++ sane-backends-1.0.6/backend/microtek2.c	Tue Nov 20 09:15:14 2001
@@ -879,8 +879,15 @@
         status = SANE_STATUS_CANCELLED;
 
     close(ms->fd[1]);
-    kill(ms->pid, SIGTERM);
-    waitpid(ms->pid, NULL, 0);
+    /* if we are aborting a scan because, for example, we run out
+     * +       of material on a feeder, then pid may be already -1 and
+     * +       kill(-1, SIGTERM), i.e. killing all our processes, is not
+     * +       likely what we really want - --mj, 2001/Nov/19 */
+    if (ms->pid > 1)
+      {
+       kill(ms->pid, SIGTERM);
+       waitpid(ms->pid, NULL, 0);
+      }
 
     return status;
 }
