--- sane-backends-1.0.20/backend/Makefile.am~	2009-03-05 02:54:32.000000000 +0100
+++ sane-backends-1.0.20/backend/Makefile.am	2010-01-19 23:34:22.344559136 +0100
@@ -94,7 +94,7 @@
 	       st400.conf stv680.conf tamarack.conf \
 	       teco1.conf teco2.conf teco3.conf test.conf \
 	       u12.conf umax1220u.conf umax.conf umax_pp.conf v4l.conf \
-	       xerox_mfp.conf dll.conf saned.conf
+	       xerox_mfp.conf dll.conf saned.conf hp_rts88xx.conf
 becfg: $(BACKEND_CONFS)
 
 SUFFIXES = .conf.in .conf
@@ -187,7 +187,7 @@
     libtamarack.la libtest.la libteco1.la \
     libteco2.la libteco3.la libu12.la libumax.la \
     libumax1220u.la libumax_pp.la libv4l.la \
-    libxerox_mfp.la
+    libxerox_mfp.la libhp_rts88xx.la
 
 # Each stand alone backend thats possible to be built should be listed
 # here.  There are the libraries that are installed under $(libdir)/sane.
@@ -218,7 +218,7 @@
     libsane-tamarack.la libsane-test.la libsane-teco1.la \
     libsane-teco2.la libsane-teco3.la libsane-u12.la libsane-umax.la \
     libsane-umax1220u.la libsane-umax_pp.la libsane-v4l.la \
-    libsane-xerox_mfp.la
+    libsane-xerox_mfp.la libsane-hp_rts88xx.la
 
 EXTRA_LTLIBRARIES = $(be_convenience_libs) $(be_dlopen_libs)
 
@@ -820,6 +820,14 @@
 # TODO: Why are these distributed but not compiled?
 EXTRA_DIST += rts8891_devices.c rts8891_low.c rts8891_low.h
 
+libhp_rts88xx_la_SOURCES = hp_rts88xx.c hp_rts88xx.h rts88xx_lib.c rts88xx_lib.h
+libhp_rts88xx_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=hp_rts88xx
+
+nodist_libsane_hp_rts88xx_la_SOURCES = hp_rts88xx-s.c
+libsane_hp_rts88xx_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=hp_rts88xx
+libsane_hp_rts88xx_la_LDFLAGS = $(DIST_SANELIBS_LDFLAGS)
+libsane_hp_rts88xx_la_LIBADD = $(COMMON_LIBS) libhp_rts88xx.la ../sanei/sanei_init_debug.lo ../sanei/sanei_constrain_value.lo ../sanei/sanei_config.lo ../sanei/sanei_config2.lo sane_strstatus.lo ../sanei/sanei_scsi.lo ../sanei/sanei_usb.lo $(SCSI_LIBS) $(USB_LIBS)
+
 libs9036_la_SOURCES = s9036.c s9036.h
 libs9036_la_CPPFLAGS = $(AM_CPPFLAGS) -DBACKEND_NAME=s9036
 
--- sane-backends-1.0.20/doc/Makefile.am~	2009-03-24 14:36:37.000000000 +0100
+++ sane-backends-1.0.20/doc/Makefile.am	2010-01-19 23:20:38.431319430 +0100
@@ -36,7 +36,7 @@
 	sane-mustek_usb2.5 sane-hp3500.5 sane-pixma.5 sane-stv680.5 \
 	sane-hp5590.5 sane-hpljm1005.5 sane-cardscan.5 sane-hp3900.5 \
 	sane-epjitsu.5 sane-hs2p.5 sane-canon_dr.5 sane-xerox_mfp.5 \
-	sane-rts8891.5 sane-coolscan3.5
+	sane-rts8891.5 sane-coolscan3.5 sane-hp_rts88xx.5
 
 EXTRA_DIST += sane-abaton.man sane-agfafocus.man sane-apple.man sane-as6e.man \
 	sane-dll.man sane-dc25.man sane-dmc.man sane-epson.man \
@@ -60,7 +60,7 @@
 	sane-pixma.man sane-stv680.man sane-hp5590.man sane-hpljm1005.man \
 	sane-cardscan.man sane-hp3900.man sane-epjitsu.man sane-hs2p.man \
 	sane-canon_dr.man sane-xerox_mfp.man sane-rts8891.man \
-	sane-coolscan3.man
+	sane-coolscan3.man sane-hp_rts88xx.man
 
 man7_MANS = sane.7
 EXTRA_DIST += sane.man
--- sane-backends-1.0.20/configure.in~	2009-05-03 03:25:23.000000000 +0200
+++ sane-backends-1.0.20/configure.in	2010-01-19 23:40:10.724520408 +0100
@@ -516,7 +516,7 @@
         pixma plustek plustek_pp qcam ricoh rts8891 s9036 \
         sceptre sharp sm3600 sm3840 snapscan sp15c st400 \
         stv680 tamarack teco1 teco2 teco3 test u12 umax 
-        umax_pp umax1220u v4l xerox_mfp"
+        umax_pp umax1220u v4l xerox_mfp hp_rts88xx"
 
 # If user specifies backends manually then cause configure
 # to fail if its detected it can't be compiled.  If we
