--- sane-backends-1.0.19/acinclude.m4.orig	2008-05-30 20:34:04.620337000 +0200
+++ sane-backends-1.0.19/acinclude.m4	2008-05-30 20:37:47.481036362 +0200
@@ -322,16 +322,6 @@
                      [use the specified group for lock dir @<:@default=uucp@:>@]),
         [LOCKPATH_GROUP="$withval"]
     )
-    # check if the group does exist
-    lasterror=""
-    touch sanetest.file
-    chgrp $LOCKPATH_GROUP sanetest.file 2>/dev/null || lasterror=$?
-    rm -f sanetest.file
-    if test ! -z "$lasterror"; then
-      AC_MSG_WARN([Group $LOCKPATH_GROUP does not exist on this system.])
-      AC_MSG_WARN([Locking feature will be disabled.])
-      use_locking=no
-    fi
   fi
   if test $use_locking = yes ; then
     INSTALL_LOCKPATH=install-lockpath
--- sane-backends-1.0.20/backend/Makefile.am~	2010-01-19 23:59:26.081372025 +0100
+++ sane-backends-1.0.20/backend/Makefile.am	2010-01-20 00:02:20.477725218 +0100
@@ -137,7 +137,7 @@
 	  done
 
 install-lockpath:
-	$(mkinstalldirs) -m 775 -g $(LOCKPATH_GROUP) $(DESTDIR)$(locksanedir)
+	$(mkinstalldirs) -m 775 $(DESTDIR)$(locksanedir)
 
 uninstall-hook:
 	rm -rf $(DESTDIR)$(libdir)/sane $(DESTDIR)$(configdir) $(DESTDIR)$(locksanedir)
