--- sane-backends-1.0.19/configure.in.orig	2008-05-30 19:01:49.213131000 +0200
+++ sane-backends-1.0.19/configure.in	2008-05-30 20:36:53.213946370 +0200
@@ -125,7 +125,7 @@
 AC_CHECK_HEADERS(fcntl.h unistd.h libc.h sys/dsreq.h sys/select.h \
     sys/time.h sys/shm.h sys/ipc.h sys/signal.h \
     sys/scanio.h scsi.h sys/scsi.h sys/scsicmd.h sys/scsiio.h \
-    bsd/dev/scsireg.h scsi/sg.h /usr/src/linux/include/scsi/sg.h \
+    bsd/dev/scsireg.h scsi/sg.h \
     camlib.h os2.h sys/socket.h sys/io.h gscdds.h sys/hw.h \
     sys/types.h sys/scsi/scsi.h sys/scsi/sgdefs.h sys/scsi/targets/scgio.h \
     apollo/scsi.h sys/sdi_comm.h sys/passthrudef.h linux/ppdev.h \
--- sane-backends-1.0.19/acinclude.m4.orig	2008-05-30 20:34:04.620337000 +0200
+++ sane-backends-1.0.19/acinclude.m4	2008-05-30 20:37:47.481036362 +0200
@@ -322,16 +322,6 @@
                      [use the specified group for lock dir @<:@default=uucp@:>@]),
         [LOCKPATH_GROUP="$withval"]
     )
-    # check if the group does exist
-    lasterror=""
-    touch sanetest.file
-    chgrp $LOCKPATH_GROUP sanetest.file 2>/dev/null || lasterror=$?
-    rm -f sanetest.file
-    if test ! -z "$lasterror"; then
-      AC_MSG_WARN([Group $LOCKPATH_GROUP does not exist on this system.])
-      AC_MSG_WARN([Locking feature will be disabled.])
-      use_locking=no
-    fi
   fi
   if test $use_locking = yes ; then
     INSTALL_LOCKPATH=install-lockpath
